libp2p/muxers/yamux/yamux.nim

```diff
@@ -377,12 +384,12 @@ proc cleanupChann(m: Yamux, channel: YamuxChannel) {.async.} =
   if channel.isReset and channel.recvWindow > 0:
     m.flushed[channel.id] = channel.recvWindow
 
-proc createStream(m: Yamux, id: uint32, isSrc: bool): YamuxChannel =
+proc createStream(m: Yamux, id: uint32, isSrc: bool, recvWindow: int): YamuxChannel =
   result = YamuxChannel(
     id: id,
-    maxRecvWindow: DefaultWindowSize,
-    recvWindow: DefaultWindowSize,
-    sendWindow: DefaultWindowSize,
+    maxRecvWindow: recvWindow,
```

Comment:
Because maxRecvWindow could be, for example, 64k but the peer you're connected to will always assume your recvWindow is 256k (or larger if you enlarge it during the SYN/ACK). In this case, the recvWindow will be larger than maxRecvWindow until it receive 192k. Then it'll not update to something larger than 64k.

Reply:
My interpretation is that other peers will always assume your initial `recvWindow` is 256k. The point seems to be if we find it valuable to allow its max size to be lower than this. I haven't seen enough evidence for that.